dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/config/ArtemisPortChecker.java

```diff
@@ -0,0 +1,95 @@
+package org.hisp.dhis.artemis.config;
+
+import org.hisp.dhis.external.conf.ConfigurationKey;
+import org.hisp.dhis.external.conf.DhisConfigurationProvider;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.net.ServerSocketFactory;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+
+/*
+ * Copyright (c) 2004-2020, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @author Morten Olav Hansen <mortenoh@gmail.com>
+ */
+@Component
+public class ArtemisPortChecker
+{
+    private final DhisConfigurationProvider dhisConfig;
+
+    public ArtemisPortChecker( DhisConfigurationProvider dhisConfig )
+    {
+        this.dhisConfig = dhisConfig;
+    }
+
+    @PostConstruct
+    public void init()
+    {
+        final int artemisPort = Integer.parseInt( dhisConfig.getProperty( ConfigurationKey.ARTEMIS_PORT ) );
+        final String artemisHost = dhisConfig.getProperty( ConfigurationKey.ARTEMIS_HOST );
+
+        if ( isEmbedded() && !isPortAvailable( artemisHost, artemisPort ) )
+        {
+            System.err.println( "\n\n" );
```

Comment:
Could we use a Slf4j logger instead of system.err? Or no point?

Reply:
It will look kinda weird? since it adds extra info... not sure, its easy enough to add it, its a bit different from normal "logging"

Reply:
I think in this specific case it does make sense to use `System.err` since the log files would be filled with data which is not really useful. It have to be quite apparent that the server can't start.

Reply:
Fair enough.